src/components/tasks/card/index.tsx

```diff
@@ -39,33 +39,37 @@ const Card: FC<Props> = ({
   const [showEditButton, setShowEditButton] = useState(false);
   const [keyLongPressed] = useKeyLongPressed();
   const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  const { actions, state } = useAppContext();
+
   useEffect(() => {
     const isAltKeyLongPressed = keyLongPressed === ALT_KEY;
     if (isAltKeyLongPressed) {
       setShowEditButton(true)
     }
   }, [keyLongPressed]);
-  useEffect(() => {
-    (async () => {
-      try{
-        const { requestPromise } = fetch(
-          { 
-            url: `${ITEM_BY_FILTER_URL}`,
-            params: {
-              itemType: 'TASK',
-              itemId: `${cardDetails.id}`
-            }
-          });
-        const { data: result } = await requestPromise
-        setTaskTagLevel(result.data)
-       } catch (err: any) {
-         toast(ERROR, err.message);
-       }
-    })()
-  },[])
   
-  const context = useAppContext() ;
-  const { actions } = context || {}
+  useEffect(() => {
+    if (state?.isLoggedIn) {
+      const getTaskTags = async () => {
```

Comment:
I feel like we could define the function irrespective of useEffect

Reply:
Added